tests/model_registry/rest_api/conftest.py

```diff
@@ -63,3 +90,272 @@ def updated_model_registry_resource(
+
+    Args:
+        admin_client: The admin client to create the ConfigMap
+        model_registry_namespace: The namespace of the model registry
+        mysql_ssl_secrets: The artifacts and secrets for the MySQL SSL connection
+
+    Returns:
+        Generator[ConfigMap, None, None]: A generator that yields the ConfigMap instance.
+    """
+    with open(mysql_ssl_artifact_paths["ca_crt"], "r") as f:
+        ca_content = f.read()
+    if not ca_content:
+        LOGGER.info("CA content is empty")
+        raise Exception("CA content is empty")
+    cm_name = "mysql-ca-configmap"
+    with ConfigMap(
+        client=admin_client,
+        name=cm_name,
+        namespace=model_registry_namespace,
+        data={"ca-bundle.crt": ca_content},
+    ) as cm:
+        yield cm
+
+
+@pytest.fixture(scope="class")
+def patch_mysql_deployment_with_ssl_ca(
+    request: pytest.FixtureRequest,
+    admin_client: DynamicClient,
+    model_registry_namespace: str,
+    model_registry_db_deployment: Deployment,
+    mysql_ssl_secrets: dict[str, Any],
+    ca_configmap_for_test: ConfigMap,
+) -> Generator[Deployment, Any, Any]:
+    """
+    Patch the MySQL deployment to use the test CA bundle (mysql-ca-configmap),
+    and mount the server cert/key for SSL.
+    """
+    CA_CONFIGMAP_NAME = request.param.get("ca_configmap_name", "mysql-ca-configmap")
+    CA_MOUNT_PATH = request.param.get("ca_mount_path", "/etc/mysql/ssl")
+    deployment = Deployment(
+        client=admin_client,
+        name=model_registry_db_deployment.name,
+        namespace=model_registry_namespace,
+    )
+    deployment.wait_for_condition(condition="Available", status="True")
+    original_deployment = deployment.instance.to_dict()
+    spec = original_deployment["spec"]["template"]["spec"]
+    my_sql_container = next(container for container in spec["containers"] if container["name"] == "mysql")
+    assert my_sql_container is not None, "Mysql container not found"
+    mysql_args = list(my_sql_container.get("args", []))
+    mysql_args.extend([
+        f"--ssl-ca={CA_MOUNT_PATH}/ca/ca-bundle.crt",
+        f"--ssl-cert={CA_MOUNT_PATH}/server_cert/tls.crt",
+        f"--ssl-key={CA_MOUNT_PATH}/server_key/tls.key",
+    ])
+
+    volumes_mounts = list(my_sql_container.get("volumeMounts", []))
+    volumes_mounts.extend([
+        {"name": CA_CONFIGMAP_NAME, "mountPath": f"{CA_MOUNT_PATH}/ca", "readOnly": True},
+        {
+            "name": "mysql-server-cert",
+            "mountPath": f"{CA_MOUNT_PATH}/server_cert",
+            "readOnly": True,
+        },
+        {
+            "name": "mysql-server-key",
+            "mountPath": f"{CA_MOUNT_PATH}/server_key",
+            "readOnly": True,
+        },
+    ])
+
+    my_sql_container["args"] = mysql_args
+    my_sql_container["volumeMounts"] = volumes_mounts
+    volumes = list(spec["volumes"])
+    volumes.extend([
+        {"name": CA_CONFIGMAP_NAME, "configMap": {"name": CA_CONFIGMAP_NAME}},
```

Comment:
May be move to a separate fixture, this fixture is doing a lot of things - setting values for different fields, and then updating the deployment. Also, why don't we create the deployment with right values? Instead of creating and patching, you can simply have a separate fixture for secure db.

Reply:
Agreed @fege, as long as this works for the time being IMHO it's ok to merge and we can improve it in a separate PR